cdap-master/src/main/java/co/cask/cdap/data/tools/ReplicationStatus.java

```diff
@@ -0,0 +1,290 @@
+      newTuple.add(replicateTime, Math.max((Long) tuple.get(replicateTime), timestamp));
+      map.put(region, newTuple);
+    }
+  }
+
+  private static ScanBuilder getScan (HBaseTableUtil tableUtil) {
+    ScanBuilder scan = tableUtil.buildScan();
+    scan.addColumn(Bytes.toBytes(Constants.ReplicationStatusTool.WRITE_TIME),
+                   Bytes.toBytes(Constants.ReplicationStatusTool.TIME));
+    scan.addColumn(Bytes.toBytes(Constants.ReplicationStatusTool.REPLICATE_TIME),
+                   Bytes.toBytes(Constants.ReplicationStatusTool.TIME));
+    scan.addColumn(Bytes.toBytes(Constants.ReplicationStatusTool.SHUTDOWN_TIME),
+                   Bytes.toBytes(Constants.ReplicationStatusTool.TIME));
+    return scan;
+  }
+
+  private static long getTimeFromResult (Result result, String time) {
+    long timestamp = 0L;
+    List<Cell> timeVal = result.getColumnCells(Bytes.toBytes(time),
+                                                Bytes.toBytes(Constants.ReplicationStatusTool.TIME));
+    if (timeVal.size() > 0) {
+      ByteBuffer timeBuffer =  ByteBuffer.wrap(CellUtil.cloneValue(timeVal.get(0)));
+      timestamp = timeBuffer.getLong();
+    }
+    return timestamp;
+  }
+
+  private static long getShutdownTime (Result shutdownTimeResult) throws IOException {
+    long shutdownTime = 0L;
+    List<Cell> shutdownVal = shutdownTimeResult.getColumnCells(
+      Bytes.toBytes(Constants.ReplicationStatusTool.SHUTDOWN_TIME),
+      Bytes.toBytes(Constants.ReplicationStatusTool.TIME));
+    if (shutdownVal.size() > 0) {
+      ByteBuffer timeBuffer = ByteBuffer.wrap(CellUtil.cloneValue(shutdownVal.get(0)));
+      shutdownTime = timeBuffer.getLong();
+      System.out.println("CDAP Shutdown Time = " + shutdownTime);
+    }
+    return shutdownTime;
+  }
+
+  /**
+   * This function implements the logic to test if replication is complete for a given table
+   * @param shutdownTime CDAP shutdown time
+   * @param writeTime timestamp of the last WAL entry written for this table
+   * @param replicateTime timestamp of the last WAL entry replicated for this table
+   * @return
+   */
+  private static boolean isReplicationComplete(long shutdownTime, long writeTime, long replicateTime) {
+    if (replicateTime > shutdownTime) {
+      // Anything after CDAP shutdown time can be ignored owing to non CDAP tables.
+      return true;
+    } else if (replicateTime == writeTime) {
+      // All entries written on the Master cluster have been replicated to the slave cluster.
+      return true;
+    }
+    return false;
+  }
+
+  private static void dumpReplicationStateTable() throws Exception {
+    HBaseTableUtil tableUtil = new HBaseTableUtilFactory(cConf).get();
+    HTable hTable = tableUtil.createHTable(hConf, getReplicationStateTableId(tableUtil));
+    ScanBuilder scan = getScan(tableUtil);
+    Result result;
+
+    System.out.println("This will dump the contents of the replicationstate table for debug purpose.");
```

Comment:
Adds the information of what is being dumped, also help if there was more added later. Reworded a bit.